R/execute.R

```diff
@@ -83,293 +123,584 @@ rmf_run_modflow.modflow <- function(modflow,
+        rui::stop("Path to {code} executable not found.")
       }
     }
     return(paste0(folder, executable))
-  } else if (name == "MODFLOW-NWT") {
+  }
+  if (grepl("nwt", code, ignore.case = TRUE)) {
+    code <- "MODFLOW-NWT"
     executable <- "MODFLOW-NWT_64.exe"
     folder <- ""
-    rmf_install_bin_folder <- paste0("C:/WRDAPP/", name, "/bin/")
+    rmf_install_bin_folder <- paste0(getOption("RMODFLOW.path"), "/", code,
+                                     "/bin/")
     if (!file.exists(executable)) {
       if (file.exists(paste0(rmf_install_bin_folder, executable))) {
         folder <- rmf_install_bin_folder
       } else if (Sys.which(executable) == "") {
-        stop("Path to ", name, " executable not found.")
+        rui::stop("Path to {code} executable not found.")
       }
     }
     return(paste0(folder, executable))
-  } else if (name == "MODFLOW-LGR") {
+  } 
+  if (grepl("lgr", code, ignore.case = TRUE)) {
+    code <- "MODFLOW-LGR"
     executable <- "mflgr.exe"
     folder <- ""
-    rmf_install_bin_folder <- paste0("C:/WRDAPP/", name, "/bin/")
+    rmf_install_bin_folder <- paste0(getOption("RMODFLOW.path"), "/", code,
+                                     "/bin/")
     if (!file.exists(executable)) {
       if (file.exists(paste0(rmf_install_bin_folder, executable))) {
         folder <- rmf_install_bin_folder
       } else if (Sys.which(executable) == "") {
-        stop("Path to ", name, " executable not found.")
+        rui::stop("Path to {code} executable not found.")
       }
     }
     return(paste0(folder, executable))
-  } else if (name == "MODFLOW-CFP") {
+  } 
+  if (grepl("cfp", code, ignore.case = TRUE)) {
+    code <- "MODFLOW-CFP"
     executable <- "mf2005cfp.exe"
     folder <- ""
-    rmf_install_bin_folder <- paste0("C:/WRDAPP/", name, "/")
+    rmf_install_bin_folder <- paste0(getOption("RMODFLOW.path"), "/", code,
+                                     "/bin/")
     if (!file.exists(executable)) {
       if (file.exists(paste0(rmf_install_bin_folder, executable))) {
         folder <- rmf_install_bin_folder
       } else if (Sys.which(executable) == "") {
-        stop("Path to ", name, " executable not found.")
+        rui::stop("Path to {code} executable not found.")
       }
     }
     return(paste0(folder, executable))
+  }
+  rui::alert("Finding paths to the executables of codes other than ",
+           "MODFLOW-2005, MODFLOW-OWHM, MODFLOW-NWT, MODFLOW-LGR or ",
+           "MODFLOW-CFP is currently not supported.")
+  rui::stop("Issue with code path.")
+}
+
+#' Look for a file path in a NAM file
+#'
+#' @param dir Character. Path to directory containing the NAM file.
+#' @param nam Character. File name of the NAM file.
+#' @param type Character. File type (ftype) of the entry to look for.
+#' @param unit Integer. Unit number of the entry to look for.
+#' @return
+rmfi_look_for_path <- function(dir, nam, type = NULL, unit = NULL) {
```

Comment:
This would also be great to use in the top-level functions!

Reply:
Yes, makes sense. Feel free to start using it! :) Maybe need to move it to the modflow.R file in that case, but that's a detail.